dns_server/src/config.rs

```diff
@@ -57,14 +57,6 @@ pub struct DnsServerConfig {
     pub mbox: Option<Name>,
 }
 
-const DEFAULT_DATA_DIR_NAME: &str = "mintlayer_dns_server";
-
 fn default_data_dir() -> String {
-    BaseDirs::new()
-        .expect("BaseDirs build failed")
-        .data_dir()
-        .join(DEFAULT_DATA_DIR_NAME)
-        .to_str()
-        .expect("expected valid default data dir path")
-        .to_owned()
+    default_data_dir_common().to_str().expect("Valid UTF-8 expected").to_owned()
```

Comment:
Is this always true? Windows uses UTF-16 in paths and registry entries.

Reply:
Good question. I looked a bit and it seems that Rust on Windows uses WTF-8 encoding internally and `to_str` succeeds most of the time (so it's not much different from on other OSs).
Anyway, I switched to `PathBuf` here.